### DISCUSSION TITLE
Possibility to "mark as read" all messages at once

### DISCUSSION THREAD

**Discussion by donsaimon** [BimmerGestalt/AAIdrive#411] [Ideas]
It would be useful if possible to mark as read all the messages at once, without having to open every single message and mark it as read.

**Comment by hufman** [comment]
This would be tricky to implement: each app would have a different action to accomplish this, and the only clue is the name, which is translated differently for each app and language, and thus is very difficult to pattern match on. If someone comes up with a clever idea, however, I'd entertain a pull request with this functionality!

**Comment by donsaimon** [comment]
For me it would be sufficient for Whatsapp

Da: ***@***.***
Inviato: 27 Settembre 2021 23:26
A: ***@***.***
Rispondi a: ***@***.***
Cc: ***@***.***; ***@***.***
Oggetto: Re: [BimmerGestalt/AAIdrive] Possibility to "mark as read" all messages at once (Discussion #411)


This would be tricky to implement: each app would have a different action to accomplish this, and the only clue is the name, which is translated differently for each app and language, and thus is very difficult to pattern match on. If someone comes up with a clever idea, however, I'd entertain a pull request with this functionality!

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/BimmerGestalt/AAIdrive/discussions/411#discussioncomment-1392799>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ASMCOD6OCZV5IV4QJH2D4ITUEDOO7ANCNFSM5ES2TFDQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Comment by bogdan-calapod** [comment]
A workaround I use that works nicely - go to the Notifications screen, hit the `Options` button and you have an option to `Remove all messages`

**Reply by donsaimon** [reply]
Ok, but in this way you only remove messages from the screen... I need to mark Whatsapp messages "as read" (2 blue "v" on sender phone) as if I have read them on the phone.

**Reply by bogdan-calapod** [reply]
@hufman would it be possible to have a `clear all` option in the notification list which goes through all notifications and:

* hits the `Mark as read` option if it exists
* dismisses the notification

I think that would be appropiate, @donsaimon ?

**Comment by donsaimon** [comment]
Yes, it could work, but the option "clear all" should be available in the main notifications page (not in the left bar), so to be quickly and easily accessible. The best thing would be the possibility to assign a shortcut button (from 1 to 8) to "clear all" function, so that it is not necessary to open the notifications page.

Da: ***@***.***
Inviato: 28 Settembre 2021 09:28
A: ***@***.***
Rispondi a: ***@***.***
Cc: ***@***.***; ***@***.***
Oggetto: Re: [BimmerGestalt/AAIdrive] Possibility to "mark as read" all messages at once (Discussion #411)


@hufman<https://github.com/hufman> would it be possible to have a clear all option in the notification list which goes through all notifications and:

  *   hits the Mark as read option if it exists
  *   dismisses the notification

I think that would be appropiate, @donsaimon<https://github.com/donsaimon> ?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/BimmerGestalt/AAIdrive/discussions/411#discussioncomment-1394428>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ASMCOD536VUZYR3KA2PJXS3UEFVDVANCNFSM5ES2TFDQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
